tests/test_helper/parallelize.bash

```diff
@@ -0,0 +1,69 @@
+# This file contains set of helper functions to run processes or functions
+# in background and collect their results.
+#
+# Caveat: Output of the processes running in parallel may overlap
+#
+# Example usage:
+# function task_distributor() {
+#     # Reserve new process pool
+#     local process_pool=""
+#     process_pool=$(new_process_pool)
+#
+#     # Start multiple commands or functions on the background
+#     start_bg_task "$process_pool" function_1 argument_1 argument_2
+#     start_bg_task "$process_pool" command_2 "$ARG_FROM_VARIABLE"
+#
+#     # Wait for all tasks in the process pool to finish
+#     gather_bg_tasks "$process_pool"
+#     # If any of the tasks failed, this function will return with return code 1
+# }
+
+declare -g -A _PROCESS_POOL
+
+# new_process_pool
+#
+# Create new process pool and print its ID.
+function new_process_pool() {
+    # Create new process pool and return its UUID
+    local new_id=""
+    new_id=$(uuidgen)
+    _PROCESS_POOL["$new_id"]=""
+    echo "$new_id"
+}
+
+# start_bg_task PROCESS_POOL FUNCTION [FUNCTION_ARGS, ...]
+#
+# Start FUNCTION in the background. Process of the executed FUNCTION will be associated
+# with the PROCESS_POOL.
+function start_bg_task() {
+    local pool_id=$1
+    shift
+    local cmd=$1
+    shift
+
+    echo "# Sending process to background: '$cmd $*'"
+    $cmd "$@" &
+    echo "# Background process has PID: $!"
+
+    _PROCESS_POOL["$pool_id"]+="$! "
+}
+
+# gather_bg_tasks PROCESS_POOL
+#
+# Wait for all processes from the PROCESS_POOL to finish. Return 1 if any of the processes
+# had non-zero RC
+function gather_bg_tasks() {
+    local pool_id=$1
```

Comment:
Add `; shift` for consistency?

Reply:
Removing unused file

Reply:
This comment can be relevant in other functions so I'm just wondering what are the benefits of using `shift` if number of arguments is known. For example this:
```bash
function verify_service_cert() {
    local container=$1
    local ip_addr=$2
    local port=$3
    ...
}
```
Looks better than:
```bash
function verify_service_cert() {
    local container=$1
    shift
    local ip_addr=$1
    shift
    local port=$1
    shift
    ...
}
```

Reply:
Oh, right, the shifts are inline. I'll push update to be inline with this style, though I'm still curious about rationale behind it.

Reply:
Thanks for updating!

As for rationale for using `var=$1; shift`:
* The lines declaring receiving variables are independent of each other, a change in the middle later affects one line and not all subsequent lines of code.
* It is quite common to have some variable receive multiple arguments, and we have to shift all previous arguments in order to use `$@` or `$*`.
* It looks more consistent when every declaration of argument receiving variables of the same type look the same.
* Our upstream projects use this style in their test suites.